### DISCUSSION TITLE
JSON errors - bad control character and missing property name

### DISCUSSION THREAD

**Discussion by Lubos-V** [rern/rAudio#1148] [Bug Reports]
Hi, I really enjoy the player. Since a couple of weeks, I have difficulties accessing part of settings. Have seen similar bugreports, but nothing addressing those specific issues. Opening the UI gives me JSON instead of output and an error message.

This is current version, updated a couple of days ago.

Thanks for any insights. I just need to know what to edit and where... ;-)

L.

### For settings / system 
Bad control character in string literal in JSON at position 748
_(possibly incorrectly parsing a tab character in fstab?)_
...
  "icon"       : "networks"
, "mountpoint" : "/mnt/MPD/NAS/servername**X**	cifs	noauto,username=USER,password=PASSWORD,uid=45,gid=45,iocharset=utf8,vers=1.0	0	0"
, "mounted"    : true

### For settings / features 
Expected property name or '}' in JSON at position 1375
_(possibly empty string?)_

, "localbrowser"     : true
, "localbrowserconf" : { **X**,"screenoff":300
,"zoom":1

**Comment by Lubos-V** [comment]
the issue remains even after the latest update. is there anything I can do myself to resolve this? i guess this is nothing major, but web page programming is not my particular strength...

**Reply by rern** [reply]
Download the latest image file and start over again.

**Reply by Lubos-V** [reply]
hi rern, 

tried that. installed a new image, but after restoring my settings the problem re-appears both in "features" and "system". not sure it´s the same place, though as the visual indicator (it was red X) is not displayed any longer, but the errors look familiar. the locations indicated are line 46 / column 26 (features) and line 26 / column 41 (system).

the problem is that i cannot do settings backup as it is not available and restore doesn´t seem to work either as it is possibly too old.

any suggestions / thoughts? do I need to start from "nothing" and reconfigure the player from scratch?

**Reply by Lubos-V** [reply]
one more thing - after launching update on the newly downloaded image, I am seeing this - it is normal?

```
:: Synchronizing package databases...
 +R downloading...
 core downloading...
error: failed retrieving file 'core.db' from Server = http: :
error: failed to synchronize all databases (download library error)
```

the playback works fine, though...

**Reply by Lubos-V** [reply]
now, after update, I am seeing the "X"s again. they are displayed in different places. interestingly enough, different devices / browsers display the JSON parser problems in different places (tried iOS/Safari, WinX/Firefox and different WinX/Edge+Chrome).

anything I can help to debug?

**Comment by findnix** [comment]
Same here.
Looks like there is a parsing error in this area:


`Errors: Unexpected token 'R', ..."ength"  : Radio Para"... is not valid JSON
`
```
, "repeat" : false
, "random" : false
, "single" : false
, "file"      : "https://stream.radioparadise.com/flac"
, "pllength"  : Radio Paradise (flac) - playlist: radioparadise.com
[playing] #1/1   1:36/0:00 (0%)
volume:100%   repeat: off   random: off   single: off   consume: off
, "song"      : 0
, "timestamp" : 1672144824333
```

`
Errors: Unexpected token 'v', ..."ength"  : volume:100"... is not valid JSON
`

```
, "repeat" : false
, "random" : false
, "single" : false
, "file"      : ""
, "pllength"  : volume:100%   repeat: off   random: off   single: off   consume: off
, "song"      : false
, "timestamp" : 1672145105881
, "stream" : true
, "Album"        : ""
, "Artist"       : ""
, "stationcover" : "/data/webradio/img/.: false
```

music control via `mpc` ist working

**Comment by findnix** [comment]
Same after install from scatch with "rAudio-1-64bit-20221117.img" and update to latest version

**Reply by Lubos-V** [reply]
i tried as well. installed the image. applied updates (without restoring configuration). after closing the update window (which had somewhat different output to what I saw in the morning), I am getting immediately:

```
Errors: Bad control character in string literal in JSON at position 1751
...
, "Artist"       : ""
, "stationcover" : "/data/webradio/img/.: false**X**
https:||stream.radioparadise.com|flac-thumb.jpg
...
```

trying to do the "from scratch" route now...

**Reply by Lubos-V** [reply]
setting up everything from scratch did the job. i don't dare trying to restore the old config...

**Comment by ottl05** [comment]
after Update to the latest version 
I´m getting:
`Errors: Unexpected token 'v', ..."ength"  : volume: n/"... is not valid JSON 
X{ 
  "player"       : "mpd"
, "btreceiver"   : false
, "card"         : 1
, "consume"      : false
, "control"      : "Digital"
, "counts"       : {
  "album": 3244,
  "albumartist": 2928,
  "artist": 3047,
  "composer": 882,
  "conductor": 0,
  "dabradio": 0,
  "date": 100,
  "genre": 183,
  "latest": 3244,
  "nas": 1,
  "playlists": 2,
  "sd": 0,
  "song": 20909,
  "usb": 0,
  "webradio": 8
}
`
---

what shall I do??

**Reply by Lubos-V** [reply]
have you restarted? I have seen some peculiar behaviour directly after update.

at the same time, similar error is emerging when I select album to be played from the library and the swipe right to get to the playback screen. JSON error. reloading the window helps, though...

**Reply by ottl05** [reply]
yes, the same problem after restart :-(

**Comment by rern** [comment]
Guys,

Please try the UPDATE branch:
Addons > lon-press Update > Ok for `UPDATE`

**Reply by ottl05** [reply]
I can´t reach the "addons - page"  :-(

**Reply by Lubos-V** [reply]
What do you mean by "Ion-press Update"? The Update button is greyed-out (I updated yesterday) - is there something else that needs to be done?

At the same time, I was struggling with the same thing before - once you are getting JSON errors in a wrong spot, then it seems to be over (at least for executing stuff via the browser). Are there command-line equivalents of what needs to be done in such a case?

**Reply by cmh714** [reply]
should read... "long-press Update button"

**Reply by ottl05** [reply]
> should read... "long-press Update button"

and how could I do this on the console per ssh?
caus due the error I´ve got no working gui in the browser

**Reply by cmh714** [reply]
hmm, not exactly sure via the cli....even though the main ui doesnt render properly, often you can still get to the other pages.....

http://your_ip_here/settings.php?p=addons

**Reply by ottl05** [reply]
ok, I made the update but still the same error

**Reply by ottl05** [reply]
I installed the actual image and made an update to the latest version. 
No problems no errors.
(202212291)

**Reply by surfninja2** [reply]
> Guys,
> 
> Please try the UPDATE branch: Addons > lon-press Update > Ok for `UPDATE`

My zero 1 got the 'unexpected token...' error. I pulled the power, booted, and was able to install 20230101 like rern suggested. 

Seems to work.

**Reply by Lubos-V** [reply]
> > should read... "long-press Update button"
> 
> and how could I do this on the console per ssh? caus due the error I´ve got no working gui in the browser

I don't have access to the system at the moment, but for the next time this is gonna happen, I copied the "cmdline script (I think it starts with curl..." that was part of the update output. The next time something like this happens, I would just paste it into the cmdline...

**Comment by cmh714** [comment]
for me, at least for the last few weeks there have been little script issues in various places...generally rebooting (sometimes several times in a row...) have gotten me past the startup issues. 

As an example of one script issue I have, using the back button in the UI when in a nested folder, causes an "error" and the back button doesnt work. 
![image](https://user-images.githubusercontent.com/8241076/209844923-edf53605-5499-4e29-bf11-6140761c0210.png)

**Reply by Lubos-V** [reply]
Have seen this as well. Not sure if it is fully reproducible but your description sounds familiar. Moving within library or changing view shows something like this. The only thing that helps is reloading the page.

**Comment by findnix** [comment]
Updated from a fresh 2022**1117** to 2022**12291** without problems.

@rern Thank You for Your outstanding work.
& Wishing you all a Happy New Year

**Comment by findnix** [comment]
Updated another player from 2022**12121** to 2022**12291** with some small problems:

```
Rearrange MPD Configuration...

ln: failed to create symbolic link '/srv/http/data/mpdconf/soxr.conf': File exist
ln: failed to create symbolic link '/srv/http/data/mpdconf/ffmpeg.conf': File exists
Job for mpd.service failed because the control process exited with error code.
See "systemctl status mpd.service" and "journalctl -xe" for details.
mpdnotrunning
MPD error: Address family not supported by protocol
MPD error: Address family not supported by protocol
/srv/http/bash/status.sh: line 161: ((: > 0 : syntax error: operand expected (error token is "> 0 ")
xset:  unable to open display ":0"

 .  Done.
```


Still had to correct `/etc/systemd/system/mpd.service.d/override.conf`
--> _ExecStart=/usr/bin/mpd **--no-daemon** /srv/http/data/mpdconf/mpd.conf_
This player runs with "Music Player Daemon 0.22.6 (0.22.6)"

Is ist a wise idea to run `pacman -Syu` or simular hence the fresh 2022117->202212291 player runs with mpd version 0.23.10?
For version 0.23.10 the "--systemd" option is valide.
